### PR TITLE
Cleanup GrpcServer and GrpcServerBuilder

### PR DESCRIPTION
**Pull request by rongrong**

### CODE REVIEW

File: core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java [hunk 1]
```diff
@@ -127,8 +87,6 @@ public boolean equals(Object o) {
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("ClientType", mClientType)
```
Comment (jja725) [inline core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java:130]: could you elaborate on why we want to remove client type and hostname here? I thought those are more for logging purpose

**Review by beinan — APPROVED** [review]
lgtm

File: core/common/src/main/java/alluxio/grpc/GrpcServerBuilder.java [hunk 2]
```diff
@@ -43,66 +41,46 @@
...
-  private AlluxioConfiguration mConfiguration;
-
-  @SuppressFBWarnings(value = "URF_UNREAD_FIELD")
-  private UserState mUserState;
+  private final AlluxioConfiguration mConfiguration;
 
   private GrpcServerBuilder(GrpcServerAddress serverAddress,
-      AuthenticationServer authenticationServer, AlluxioConfiguration conf, UserState userState) {
-    mAuthenticationServer = authenticationServer;
+      AuthenticationServer authenticationServer, AlluxioConfiguration conf) {
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/grpc/GrpcServerBuilder.java:55]: AuthenticationServer is nullable? Use optional?

**Review by LuQQiu — APPROVED** [review]
LGTM, left one minor comment

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**